src/test/java/pl/byczazagroda/trackexpensesappbackend/service/UserServiceImplTest.java

```diff
@@ -0,0 +1,114 @@
+package pl.byczazagroda.trackexpensesappbackend.service;
+
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import pl.byczazagroda.trackexpensesappbackend.dto.AuthRegisterDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.AppRuntimeException;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.model.User;
+import pl.byczazagroda.trackexpensesappbackend.repository.UserRepository;
+
+import static org.junit.Assert.*;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.any;
+import org.mockito.ArgumentCaptor;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+public class UserServiceImplTest {
+
+    private static final AuthRegisterDTO REGISTER_DTO =
+            new AuthRegisterDTO("test@test.com", "Test123@", "testuser");
+
+    private static final AuthRegisterDTO REGISTER_DTO_TOO_SHORT_PASSWORD =
+            new AuthRegisterDTO("test@test.com", "short", "testuser");
+
+    private static final AuthRegisterDTO REGISTER_DTO_INVALID_EMAIL =
+            new AuthRegisterDTO("InvalidEmail", "Test123@", "testuser");
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserServiceImpl userService;
+
+    @BeforeEach
+    public void setUp() {
+        Mockito.reset(userRepository);
+    }
+
+    @DisplayName("When password is valid, it should return a hashed password of correct length")
+    @Test
+    public void testHashPassword_WhenPasswordIsValid_ThenReturnHashed() {
```

Comment:
like in your integration tests When and Then should be lowercase so
from `testHashPassword_WhenPasswordIsValid_ThenReturnHashed`
to `testHashPassword_whenPasswordIsValid_thenReturnHashed`
applies to whole class